File: src/main/java/io/github/svegon/capi/event/ListenerSet.java
package io.github.svegon.capi.event;

import java.util.Set;
import java.util.function.Function;

public class ListenerSet<L> extends ListenerCollection<L, Set<L>> {
    ListenerSet(Set<L> listeners, Function<Set<L>, L> invokerFactory) {
        super(listeners, invokerFactory);
    }
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/listener_collections/ListenerSet.kt
package io.github.svegon.mclientapi.event.listener_collections

import java.util.*

class ListenerSet<L> internal constructor(listeners: MutableSet<L>, invokerFactory: (Set<L>) -> L) :
    ListenerCollection<L, MutableSet<L>, Set<L>>(listeners, invokerFactory) {
    override fun makeUnmodifiable(listeners: MutableSet<L>): Set<L> {
        return Collections.unmodifiableSet(listeners)
    }
}
